core/src/sumeragi/main_loop.rs

```diff
@@ -624,7 +652,14 @@ fn process_message_independent(
                     let event_recommendations = Vec::new();
                     let new_block = match BlockBuilder::new(
                         transactions,
-                        sumeragi.current_topology.clone(),
+                        Topology::recreate_topology(
+                            &sumeragi
+                                .wsv
+                                .latest_block_ref()
+                                .expect("WSV must have blocks"),
+                            current_view_change_index,
+                            current_topology.ordered_peers.iter().cloned().collect(),
```

Comment:
Can we just grab peers from `wsv`?

Reply:
Yes, they are the same. This makes more sense to me.